test/e2e_node/runtime_conformance_test.go

```diff
@@ -56,259 +56,129 @@ var _ = Describe("[FLAKY] Container runtime Conformance Test", func() {
 	})
 
 	Describe("container runtime conformance blackbox test", func() {
-		var testCContainers []ConformanceContainer
 		namespace := "runtime-conformance"
 
-		BeforeEach(func() {
-			testCContainers = []ConformanceContainer{}
-		})
-
-		Context("when start a container that exits successfully", func() {
+		Context("when starting a container that exits", func() {
 			It("it should run with the expected status [Conformance]", func() {
+				restartCountVolumeName := "restart-count"
+				restartCountVolumePath := "/restart-count"
 				testContainer := api.Container{
 					Image: ImageRegistry[busyBoxImage],
 					VolumeMounts: []api.VolumeMount{
 						{
-							MountPath: "/restart-count",
-							Name:      "restart-count",
+							MountPath: restartCountVolumePath,
+							Name:      restartCountVolumeName,
 						},
 					},
-					ImagePullPolicy: api.PullIfNotPresent,
 				}
 				testVolumes := []api.Volume{
 					{
-						Name: "restart-count",
+						Name: restartCountVolumeName,
 						VolumeSource: api.VolumeSource{
 							HostPath: &api.HostPathVolumeSource{
 								Path: os.TempDir(),
 							},
 						},
 					},
 				}
-				testCount := int32(3)
-				testStatuses := []testStatus{
-					{"terminate-cmd-rpa", api.RestartPolicyAlways, api.PodRunning, ContainerStateWaiting | ContainerStateRunning | ContainerStateTerminated, ">", testCount, false},
-					{"terminate-cmd-rpof", api.RestartPolicyOnFailure, api.PodSucceeded, ContainerStateTerminated, "==", testCount, false},
-					{"terminate-cmd-rpn", api.RestartPolicyNever, api.PodSucceeded, ContainerStateTerminated, "==", 0, false},
+				testCases := []testCase{
+					{"terminate-cmd-rpa", api.RestartPolicyAlways, api.PodRunning, ContainerStateRunning, "==", 2, true},
+					{"terminate-cmd-rpof", api.RestartPolicyOnFailure, api.PodSucceeded, ContainerStateTerminated, "==", 1, false},
+					{"terminate-cmd-rpn", api.RestartPolicyNever, api.PodFailed, ContainerStateTerminated, "==", 0, false},
 				}
-
-				for _, testStatus := range testStatuses {
+				for _, testCase := range testCases {
 					tmpFile, err := ioutil.TempFile("", "restartCount")
 					Expect(err).NotTo(HaveOccurred())
 					defer os.Remove(tmpFile.Name())
 
-					// It fails in the first three runs and succeeds after that.
-					tmpCmd := fmt.Sprintf("echo 'hello' >> /restart-count/%s ; test $(wc -l /restart-count/%s| awk {'print $1'}) -ge %d", path.Base(tmpFile.Name()), path.Base(tmpFile.Name()), testCount+1)
-					testContainer.Name = testStatus.Name
+					// It fails in the first 2 runs and succeeds after that.
```

Comment:
Should be: Failed in the first run, then succeeded at the second one, then run forever?